src/os/ObjectStore.h

```diff
@@ -1358,6 +1363,36 @@ class ObjectStore {
       }
       data.ops++;
     }
+
+   /**
+    * Move temporary object to base object.
+    * Data portion is only copied from temp object to base object.
+    * The copy is done according to the move_info vector of tuple, which
+    * has information of src_offset, dest_offset and length.
+    * Finally, the temporary object is deleted.
+    */
+    void move_tempobjs(const coll_t& temp_cid, const ghobject_t& temp_oid, coll_t cid,
```

Comment:
Also, we don't need to pass source/temp_cid here at all.  At the ObjectStore level, you can't do a move between collections anyway (the filestore might kludge the source or dest collection to the temp collection if it needs to).